test/parallel/test-whatwg-url-parsing.js

```diff
@@ -22,6 +22,7 @@ const failureTests = tests.filter((test) => test.failure).concat([
   { input: null },
   { input: new Date() },
   { input: new RegExp() },
+  { input: 'test', base: null},
```

Comment:
Sorry, my example in the issue was not very good. This throws because `'test'` is not a valid URL. With this change, a valid input with a base of `null` will not throw. To make it work, we still need to reach the line with `base = new URL(base);`

Reply:
Thanks for clarifying!

Reply:
Added two relevant examples below

Reply:
@targos please take another look - I've changed the code since that review.